Modify radial buffer regions

bendudson opened the pull request:
- Remove averaging in Y, except in the core closed field-line region.
- Add a switch radial_core_averagey to turn on/off this averaging in the core
- Modify treatment of NVi to be similar to the other fields, rather than damping
  to zero.

bendudson committed:
Modify radial buffer regions

- Remove averaging in Y, except in the core closed field-line region.
- Add a switch radial_core_averagey to turn on/off this averaging in the core
- Modify treatment of NVi to be similar to the other fields, rather than damping
  to zero.

bendudson committed:
Fix copy-paste error in outer radial buffer

Thanks @hahahasan ! Accidentally using Vort rather than NVi in
outer boundary.

johnomotani commented on hermes-2.cxx:
```diff
@@ -3384,11 +3388,21 @@ int Hermes::rhs(BoutReal t) {
+    Field2D PiInner = radial_inner_averagey ? averageY(PiDC) : PiDC;
+    Field2D NeInner = radial_inner_averagey ? averageY(NeDC) : NeDC;
+    Field2D VortInner = radial_inner_averagey ? averageY(VortDC) : VortDC;
+    Field2D NViInner = radial_inner_averagey ? averageY(NViDC) : NViDC;
```
Do you want to have an option to damp `NVi` toward zero on a core boundary, like you had before this PR? Could add another input flag like `zero_inner_NVi` and set `NViInner` to 0.0 if it's true.

bendudson committed:
Add options for averaging in Y in core buffer region

- Turn off averaging of vorticity and NVi by default, since this seems to
  lead to problems, and there's no reason these would be
  expected to be constant on a core flux surface.

- Add option to damp NVi towards zero on core boundary
